tests/src/main/java/com/hazelcast/stabilizer/tests/map/MapHeapHogTest.java

```diff
@@ -1,116 +1,149 @@
 package com.hazelcast.stabilizer.tests.map;
 
-
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.core.IMap;
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.logging.Logger;
 import com.hazelcast.stabilizer.test.TestContext;
-import com.hazelcast.stabilizer.test.annotations.Run;
+import com.hazelcast.stabilizer.test.annotations.RunWithWorker;
 import com.hazelcast.stabilizer.test.annotations.Setup;
 import com.hazelcast.stabilizer.test.annotations.Verify;
 import com.hazelcast.stabilizer.test.annotations.Warmup;
-import com.hazelcast.stabilizer.test.utils.ThreadSpawner;
+import com.hazelcast.stabilizer.worker.tasks.AbstractMonotonicWorkerTask;
+import com.hazelcast.stabilizer.worker.tasks.AbstractWorkerTask;
 
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import static com.hazelcast.stabilizer.test.utils.TestUtils.humanReadableByteCount;
 import static com.hazelcast.stabilizer.tests.helpers.HazelcastTestUtils.isMemberNode;
 import static com.hazelcast.stabilizer.tests.helpers.HazelcastTestUtils.nextKeyOwnedBy;
+import static com.hazelcast.stabilizer.utils.CommonUtils.sleepSeconds;
+import static java.lang.String.format;
 
 public class MapHeapHogTest {
 
-    private final static ILogger log = Logger.getLogger(MapHeapHogTest.class);
+    private static final int MIN_CLUSTER_SIZE = 3;
+    private static final long APPROX_ENTRY_BYTES_SIZE = 238;
+    private static final ILogger LOGGER = Logger.getLogger(MapHeapHogTest.class);
 
     // properties
-    public String basename = this.getClass().getName();
+    public String basename = this.getClass().getSimpleName();
     public int threadCount = 3;
     public int ttlHours = 24;
     public double approxHeapUsageFactor = 0.9;
+    public int logFrequency = 500000;
+    /**
+     * Configures the worker threads to use the same keys, which slows down the fill rate while keeping the load high.
+     */
+    public boolean useSameKeysOnAllWorker = true;
```

Comment:
Let's make this easy: You want to fill the cluster map with data (by putting local keys into the map).

Should the test do this as fast as possible or by creating the most possible load?

Speed: all local worker threads should use different keys (fast + parallel puts)
Load: all local worker threads should use the same keys (slow + parallel puts)